tests/test_tokenization_common.py

```diff
@@ -3330,6 +3336,31 @@ def test_tokenizer_mismatch_warning(self):
                             )
                         )
 
+    @require_torch
+    def test_saving_tokenizer_trainer(self):
+        for tokenizer, pretrained_name, kwargs in self.tokenizers_list:
+            with self.subTest(f"{tokenizer.__class__.__name__} ({pretrained_name})"):
+                with tempfile.TemporaryDirectory() as tmp_dir:
+                    # Save the fast tokenizer files in a temporary directory
+                    tokenizer_old = self.rust_tokenizer_class.from_pretrained(pretrained_name, **kwargs, use_fast=True)
+                    tokenizer_old.save_pretrained(tmp_dir, legacy_format=False)  # save only fast version
+
+                    # Initialize toy model for the trainer
+                    model = nn.Module()
+
+                    # Load tokenizer from a folder without legacy files
+                    tokenizer = self.rust_tokenizer_class.from_pretrained(tmp_dir)
+                    training_args = TrainingArguments(
+                        output_dir=tmp_dir,
+                        do_train=True,
+                        no_cuda=True,
+                    )
+                    trainer = Trainer(model=model, args=training_args, tokenizer=tokenizer)
+
+                    # Should not raise an error
+                    trainer.save_model(os.path.join(tmp_dir, "checkpoint"))
+                    assert "tokenizer.json" in os.listdir(os.path.join(tmp_dir, "checkpoint"))
+
```

Comment:
I finally chose to add a new test to all tokenizers by adding this test to `test_tokenization_common.py` instead of the previous test in `test_trainer.py`.